content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/system-usage.md

```diff
@@ -0,0 +1,85 @@
+
+### The `terrainExists` flag
+
+Because terrain data only exists within arbitrary authored regions, it's possible to query the terrain system for points where the terrain does not exist. The terrain is defined as existing as long as height data exists. When querying for terrain data, this flag should always be checked for each query result to ensure that the data actually represents valid terrain.
+
+### Query optimizations
+
+Here are a few tips for getting the best performance out of the terrain query APIs:
+
+* The APIs are designed to let you query for different combinations of height, normal, and surface data. The more data you query, the more expensive the query will be, so try to limit queries to fetch the minimum amount of data that you intend to use. When you need multiple types of data for the same position, the APIs that let you fetch multiple types at once will generally be faster than calling the individual APIs separately. For example, `GetSurfacePoint` will be faster than calling `GetHeight`, `GetNormal`, and `GetSurfaceWeights` individually.
+* When querying for multiple input positions, the bulk APIs (`QueryList`, `QueryRegion`) are generally 80-90% faster than calling the individual APIs (`GetHeight`, etc).
+* `QueryRegion` / `QueryRegionAsync` are the fastest query APIs because they don't require the additional step of building up an input list of positions before calling them.
+* The queries can be made up to 4x faster by using the `Exact` or `Clamp` sampler choices, as long as you take care to align the input positions or input region exactly to the terrain grid. Misalignment will produce inaccurate results.
+* The `TerrainAreaExistsInBounds` provides a quick way to determine if terrain exists at _all_ within an AABB. This can be used to circumvent the more expensive per-point queries entirely if the region your system cares about doesn't contain any terrain.
+
+### Raycast queries
+
+The `TerrainDataRequestBus` provides a `GetClosestIntersection` method to perform a single raycast against the terrain. Typically, collision raycasts are expected to be performed through a physics system, but this method is available as an alternative for when physics raycasts aren't appropriate. For example, it's used within the Editor itself for placing objects on the terrain.
+
+If your system is always raycasting straight down to get terrain information directly beneath a location, the query APIs will produce faster results than the raycast API.
```

Comment:
I'm assuming the query APIs refers to the ones mentioned earlier on this page, and the raycast API refers to this `GetClosestIntersection` function? If so, I suggest adding that clarity here. I think it starts to get confusing which API you're talking about because this section is called "raycast queries"

```suggestion
If your system often raycasts straight down to get terrain information directly beneath a location, the query APIs that are mentioned earlier produces faster results than this raycast API.
```

Reply:
Hm, after re-reading this block, how about this instead:

```suggestion
{{< tip >}}
`GetClosestIntersection` provides generalized raycasts in any direction, but if your system only raycasts straight down to get terrain information directly beneath a location, use one of the other query APIs such as `GetSurfacePoint` or `QueryList` instead. They produce faster results.
{{< /tip >}}
```

Reply:
Yes that sounds clearer! Small note: "generalized" -> "general"

Reply:
Discussed offline, we'll keep "generalized" since it has an intentional use in this sentence.